File: __iOS_BACKUP__/ShapeByte/features/workout/schedule/domain/CurrentWorkoutScheduleEntryUseCase.swift
//
//  CurrentWorkoutScheduleEntryUseCase.swift
//  ShapeByte
//
//  Created by Lang, Stefan [Shape Byte Tech] on 02.08.24.
//

import Foundation

class CurrentWorkoutScheduleEntryUseCase {
    private let repository: WorkoutScheduleRepository

    init(repository: WorkoutScheduleRepository) {
        self.repository = repository
    }

    func invoke() -> WorkoutScheduleEntry? {
        return repository.currentScheduleEntry()
    }
}

File: features/workout/api/src/commonMain/kotlin/de/stefan/lang/shapebyte/features/workout/api/schedule/CurrentWorkoutScheduleEntryUseCase.kt
package de.stefan.lang.shapebyte.features.workout.api.schedule

import de.stefan.lang.foundation.core.contract.loadstate.LoadState
import de.stefan.lang.foundation.core.contract.usecase.BaseDataUseCase
import de.stefan.lang.utils.logging.contract.Logging
import kotlinx.coroutines.flow.Flow

abstract class CurrentWorkoutScheduleEntryUseCase(logger: Logging) :
    BaseDataUseCase<WorkoutScheduleEntry?>(logger) {
    abstract operator fun invoke(): Flow<LoadState<WorkoutScheduleEntry?>>
}
